### PR TITLE
Upgrade max char

### PR DESCRIPTION
**Pull request by fahimfaisaal**
# Welcome to the JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Replace the recent algorithm.
- [x] Replace the test case.
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: String/MaxCharacter.js [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
+    occurrenceMap.set(char, occurrenceMap.get(char) + 1 || 1)
   }
-  let maxCharacter = null
-  let maxCount = 0
-  Object.keys(occurrences).forEach(char => {
-    if (occurrences[char] > maxCount) {
-      maxCount = occurrences[char]
-      maxCharacter = char
+
+  let max = { char: '', occur: -Infinity }
```
Comment (appgurueu) [inline String/MaxCharacter.js:21]: Why store both? I'd much prefer this being replaced with a `reduce` BTW

**Review by appgurueu — CHANGES_REQUESTED** [review]
To be fair I only see a minor, questionable improvement (the customizable character filter) and a bunch of extraneous syntactic sugar.

**Comment by fahimfaisaal** [comment]
I feel now it's prefect

**Review by appgurueu — CHANGES_REQUESTED** [review]
Please fix the raised issues

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**